src/services/decoders/JsonlDecoder/index.ts

```diff
@@ -122,6 +122,29 @@ class JsonlDecoder implements Decoder {
         return results;
     }
 
+    getLogEventIdxByTimestamp (timestamp: number): number {
+        let low = 0;
+        let high = this.#logEvents.length - 1;
+        let result = -1;
+
+        while (low <= high) {
+            const mid = Math.floor((low + high) / 2);
+
+            // @ts-expect-error TS2532: Object is possibly 'undefined'.
+            const midTimestamp = this.#logEvents[mid].timestamp.valueOf();
```

Comment:
Since `@ts-expect-error` disables type-checking for the whole following line, can we extract a `const midLogEvent = this.#logEvents[mid];` and only disable type-checking for that?

Reply:
How about we use the ! to make assertions and disable eslint for using non-null assertions